sealtk/core/test/TestTracks.cpp

```diff
@@ -29,15 +29,26 @@ namespace test
 
 // ----------------------------------------------------------------------------
 void testTrackData(QAbstractItemModel const& model,
-                   QModelIndex const& index, QRectF const& box)
+                   QModelIndex const& index, TrackState const& state)
 {
   QVERIFY(model.data(index, AreaLocationRole).canConvert<QRectF>());
-  QCOMPARE(model.data(index, AreaLocationRole).value<QRectF>(), box);
+  QCOMPARE(model.data(index, AreaLocationRole).toRectF(),
+           state.location);
+
+  auto const modelHasClassification =
+    !model.data(index, ClassificationRole).isValid();
+  if (state.classification.isEmpty() == !modelHasClassification)
```

Comment:
Shouldn't we be asserting that these two are equal, and then doing the contents of the if-block if `modelHasClassification` is true?

Reply:
Uh... something like that. Please see if the updated version is better.

(If there is no classification, both `QVariant{}` and `QVariant{QVariantHash{}}` are acceptable results.)